helm/ingress-azure/templates/deployment.yaml

```diff
@@ -110,11 +110,16 @@ spec:
           path: /etc/kubernetes/
           type: Directory
       {{- if .Values.armAuth }}
-      {{- if eq .Values.armAuth.type "servicePrincipal"}}
+      {{- if and (eq .Values.armAuth.type "servicePrincipal") (not .Values.armAuth.existingSecret) }}
```

Comment:
```yaml
      {{- if eq .Values.armAuth.type "servicePrincipal"}}
      - name: networking-appgw-k8s-azure-service-principal-mount
        secret:
          secretName: networking-appgw-k8s-azure-service-principal
      {{- else if .Values.armAuth.existingSecret }}
      - name: networking-appgw-k8s-azure-service-principal-mount
        secret:
          secretName: {{ .Values.armAuth.existingSecret }}
      {{- end}}
```

Reply:
Thanks for your review.

Not sure this makes sense, the Auth Type will be servicePrincipal but we have an already existing secret, this way I would need to set the armAuthType to something else and set the existingSecret only. Or do I understand it wrong?